cli_tools/gce_vm_image_import/importer/data_disk_processor.go

```diff
@@ -53,14 +53,18 @@ func newDataDiskProcessor(pd persistentDisk, client createImageClient, project s
 	}
 }
 
-func (d dataDiskProcessor) traceLogs() []string {
+func (d *dataDiskProcessor) traceLogs() []string {
 	return []string{}
 }
 
-func (d dataDiskProcessor) process(ctx context.Context) (err error) {
+func (d *dataDiskProcessor) process(ctx context.Context) (err error) {
 	return d.client.CreateImage(d.project, &d.request)
 }
 
+func (d *dataDiskProcessor) cancel(reason string) {
+	//not possible to cancel direct calls to compute client and not likely to take long
```

Comment:
If it's a long API call (like create instance/disk), there is a way to cancel techincally. The "operation" id can be used to cancel.

Reply:
Do you have any examples?

Reply:
Here: https://github.com/GoogleCloudPlatform/compute-image-tools/blob/master/daisy/compute/compute.go#L394
The c.raw.Disks.Insert actually returns a operation, which is a handle and can be used to cancel a long API.
Currently we don't have a mechanism to cancel it. If we want, we need to add some mechanism in daisy.Compute. You can decide whether it's worth doing it for this project.

Reply:
How about instantiating the compute client with a context that has the deadline applied? Or switching to a client that takes the context as a parameter?

Reply:
Based on my observations, when context expires, compute client will lose connectivity to the API endpoint. So having context expire will not do much for cancelling an operation. Instead, we would be unable to do any further clean-ups.

For this particular case, since image creation is fast, what we can do is wait for the operation to finish and, if it succeeded, delete the image that was created. 

I don't have any experience with cancelling compute operations, i.e. not sure how reliable it is. If we cancel an image create, are we guaranteed image will never be created? If not, we might still need to do a check ourselves and delete the image in our code.

Reply:
As I think about it, I like your current approach here. Given these two options, it seems like the first is the most user friendly (at leas to me!):
1. Exceed the timeout budget by a few seconds, but successfully finish the import and create an image.
2. Strictly follow the timeout budget, and delete the final image that would otherwise have been successful. 

A small irony in the second approach, too, is that deleting the image would require a nonzero amount of time :)

Reply:
I've implemented the option 1. from Eric's comment: if timeout happens during data disk processor step, wrapper will let it finish and not return timeout error. Since image create takes around a minute in my tests, with the extra buffer we have to perform cleanup, this should not lead to many terminated imports and will provide better customer experience with less failures that could have succeeded. I'll monitor the percentage of STARTED imports (which would indicate Cloud Build times out before the wrapper) and adjust the logic if needed.